src/descriptor/bare.rs

```diff
@@ -219,7 +220,7 @@ impl<Pk: MiniscriptKey> Pkh<Pk> {
     /// sighash suffix. Includes the weight of the VarInts encoding the
     /// scriptSig and witness stack length.
     pub fn max_satisfaction_weight(&self) -> usize {
-        4 * (1 + 73 + BareCtx::pk_len(&self.pk))
+        4 * (1 + 73 + BareCtx::pk_len(&KeyExpr::SingleKey(self.pk.clone())))
```

Comment:
Let's convert back `pk_len` to accept a `&Pk` so that we avoid the clone here

Reply:
Let's add a test for Checking the len of 1) `pk(uncompressedkey)` in Legacy. 2) pk(`compressed`) in Segwitv0 3) pk(`XonlyKey`) in Tap and 4) pk('musig') in tap

Reply:
I have posted what to change in the comment at the function definition in the context.rs file.

Reply:
1. Revert back to &Pk.
2. Add a test `check_script_size` covering all the four suggested tests.